### PR TITLE
Subscriptions

### PR DESCRIPTION
**Pull request by kunalkushwah**
Adds support for [Razorpay Subscriptions](https://razorpay.com/subscriptions/)

### CODE REVIEW

File: razorpay/resources/addon.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from .base import Resource
+from ..constants.url import URL
+import warnings
```
Comment (gaurav1068) [inline razorpay/resources/addon.py]: unused import

File: razorpay/resources/plan.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from .base import Resource
+from ..constants.url import URL
+import warnings
```
Comment (gaurav1068) [inline razorpay/resources/plan.py]: unused

File: razorpay/resources/plan.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    def all(self, data={}, **kwargs):
+        """"
+        Fetch all plan entities
+
+        Returns:
+            Dictionary of plan data
+        """
+        return super(Plan, self).all(data, **kwargs)
+
```
Comment (gaurav1068) [inline razorpay/resources/plan.py]: no new line at end of file

File: razorpay/resources/subscription.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+from .base import Resource
+from ..constants.url import URL
+import warnings
```
Comment (gaurav1068) [inline razorpay/resources/subscription.py]: unused

File: razorpay/resources/subscription.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            Dictionary of Subscription data
+        """
+        return super(Subscription, self).all(data, **kwargs)
+
+    def fetch(self, subscription_id, data={}, **kwargs):
+        """"
+        Fetch Subscription for given Id
+
+        Args:
+            subscription_id : Id for which subscription object has to be retrieved
```
Comment (gaurav1068) [inline razorpay/resources/subscription.py]: try to break into multiple line at L37 also

File: razorpay/resources/subscription.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        Create addon for given subscription
+
+        Args:
+            subscription_id : Id for which addon has to be created
+
+        Return:
+            Subscription dict for given subscription id
+        """
+        url = "{}/{}/addons".format(self.base_url, subscription_id)
+        return self.post_url(url, data, **kwargs)
```
Comment (gaurav1068) [inline razorpay/resources/subscription.py]: no new line at end

File: tests/mocks/plan_collection.json [hunk 7]
```diff
@@ -0,0 +1,170 @@
+{
+	"count": 7,
+	"items": [{
+		"created_at": 1507032151,
```
Comment (harman28) [inline tests/mocks/plan_collection.json]: Indentation and spacing here is different from other files

File: tests/mocks/plan_collection.json [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+		},
+		"id": "plan_8aoXJnOqQAhq55"
+	}, {
+		"created_at": 1503394895,
+		"notes": [],
+		"interval": 1,
+		"period": "monthly",
+		"entity": "plan",
+		"item": {
+			"description": "dummy . \\n   really.. nothing great!!!",
```
Comment (harman28) [inline tests/mocks/plan_collection.json]: Kya hai yeh

File: tests/mocks/plan_collection.json [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+	}, {
+		"created_at": 1504869309,
+		"notes": {
+			"undefined": "random"
+		},
+		"interval": 1,
+		"period": "monthly",
+		"entity": "plan",
+		"item": {
+			"description": null,
```
Comment (harman28) [inline tests/mocks/plan_collection.json]: Arrange keys in the order we actually receive them in the API.

File: tests/mocks/subscription_collection.json [hunk 10]
```diff
@@ -0,0 +1,176 @@
+{
+   "count": 9,
+   "items": [{
```
Comment (harman28) [inline tests/mocks/subscription_collection.json]: Use a prettifier on all of these samples

File: tests/mocks/subscription_collection.json [hunk 11]
```diff
@@ -0,0 +1,176 @@
+{
+   "count": 9,
```
Comment (harman28) [inline tests/mocks/subscription_collection.json]: Just 2 would do too :)

File: tests/mocks/plan_collection.json (hunk 8 above)
Comment (gaurav1068) [reply tests/mocks/plan_collection.json]: lol

**Pull request merged**